contracts/libs/Witnet.sol

```diff
@@ -3,314 +3,371 @@
-  function asInt128Array(WitnetData.Result memory _result) external pure returns(int128[] memory) {
-    require(_result.success, "Tried to read `int128[]` value from errored WitnetData.Result");
-    return _result.value.decodeInt128Array();
-  }
-
-  /// @notice Decode a string value from a WitnetData.Result as a `string` value.
-  /// @param _result An instance of WitnetData.Result.
-  /// @return The `string` decoded from the WitnetData.Result.
-  function asString(WitnetData.Result memory _result) external pure returns(string memory) {
-    require(_result.success, "Tried to read `string` value from errored WitnetData.Result");
-    return _result.value.decodeString();
-  }
-
-  /// @notice Decode an array of string values from a WitnetData.Result as a `string[]` value.
-  /// @param _result An instance of WitnetData.Result.
-  /// @return The `string[]` decoded from the WitnetData.Result.
-  function asStringArray(WitnetData.Result memory _result) external pure returns(string[] memory) {
-    require(_result.success, "Tried to read `string[]` value from errored WitnetData.Result");
-    return _result.value.decodeStringArray();
-  }
-
-  /// @notice Decode a natural numeric value from a WitnetData.Result as a `uint64` value.
-  /// @param _result An instance of WitnetData.Result.
-  /// @return The `uint64` decoded from the WitnetData.Result.
-  function asUint64(WitnetData.Result memory _result) external pure returns(uint64) {
-    require(_result.success, "Tried to read `uint64` value from errored WitnetData.Result");
-    return _result.value.decodeUint64();
-  }
+    /// Data kept in EVM-storage for every Request posted to the Witnet Request Board.
+    struct Request {
+        IWitnetRequest addr;    // The contract containing the Data Request which execution has been requested.
+        address requestor;      // Address from which the request was posted.
+        bytes32 codehash;       // Codehash of the Data Request which execution has been requested.
+        uint256 gasprice;       // Minimum gas price the DR resolver should pay on the solving tx.
+        uint256 reward;         // escrow reward to by paid to the DR resolver.
+    }
 
-  /// @notice Decode an array of natural numeric values from a WitnetData.Result as a `uint64[]` value.
-  /// @param _result An instance of WitnetData.Result.
-  /// @return The `uint64[]` decoded from the WitnetData.Result.
-  function asUint64Array(WitnetData.Result memory _result) external pure returns(uint64[] memory) {
-    require(_result.success, "Tried to read `uint64[]` value from errored WitnetData.Result");
-    return _result.value.decodeUint64Array();
-  }
+    /// Data kept in EVM-storage containing Witnet-provided response metadata and result.
+    struct Response {
+        address reporter;       // Address from which the result was reported.
+        uint256 timestamp;      // EVM-provided timestamp in which the result was reported. 
```

Comment:
```suggestion
        uint256 timestamp;      // EVM-provided timestamp at which the result was reported. 
```

Because timestamps are specific times, much equivalent to stating hours and minutes: https://www.eurocentres.com/blog/talking-about-time-using-at-in-and-on